fdbserver/BlobManager.actor.cpp

```diff
@@ -1314,6 +1320,585 @@ ACTOR Future<Void> blobWorkerRecruiter(
+				wait(tr.commit());
+			}
+			break;
+		} catch (Error& e) {
+			printf("Attempt to clear prune intent got error %s\n", e.name());
+			wait(tr.onError(e));
+		}
+	}
+
+	if (BM_DEBUG) {
+		printf("Successfully pruned range [%s-%s) at pruneVersion=%lld\n",
+		       startKey.printable().c_str(),
+		       endKey.printable().c_str(),
+		       pruneVersion);
+	}
+	return Void();
+}
+
+/*
+ * This monitor watches for changes to a key K that gets updated whenever there is a new prune intent.
+ * On this change, we scan through all blobGranulePruneKeys (which look like <startKey, endKey>=<prune_version,
+ * force>) and prune any intents.
+ *
+ * Once the prune has succeeded, we clear the key IF the version is still the same one that was pruned.
+ * That way, if another prune intent arrived for the same range while we were working on an older one,
+ * we wouldn't end up clearing the intent.
+ *
+ * When watching for changes, we might end up in scenarios where we failed to do the work
+ * for a prune intent even though the watch was triggered (maybe the BM had a blip). This is problematic
+ * if the intent is a force and there isn't another prune intent for quite some time. To remedy this,
+ * if we don't see a watch change in X (configurable) seconds, we will just sweep through the prune intents,
+ * consolidating any work we might have missed before.
+ *
+ * Note: we could potentially use a changefeed here to get the exact pruneIntent that was added
+ * rather than iterating through all of them, but this might have too much overhead for latency
+ * improvements we don't really need here (also we need to go over all prune intents anyways in the
+ * case that the timer is up before any new prune intents arrive).
+ */
+ACTOR Future<Void> monitorPruneKeys(BlobManagerData* self) {
+	try {
+		state Value oldPruneWatchVal;
+		loop {
+			state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(self->db);
+			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+
+			// Wait for the watch to change, or some time to expire (whichever comes first)
+			// before checking through the prune intents. We write a UID into the change key value
+			// so that we can still recognize when the watch key has been changed while we weren't
+			// monitoring it
+			loop {
+				try {
+					tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+					tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+
+					state Optional<Value> newPruneWatchVal = wait(tr->get(blobGranulePruneChangeKey));
+
+					// if the value at the change key has changed, that means there is new work to do
+					if (newPruneWatchVal.present() && oldPruneWatchVal != newPruneWatchVal.get()) {
+						oldPruneWatchVal = newPruneWatchVal.get();
+						if (BM_DEBUG) {
+							printf("the blobGranulePruneChangeKey changed\n");
+						}
+
+						// TODO: debugging code, remove it
```

Comment:
Will remove before merging. Please ignore the `dummy` txn

Reply:
Commented out instead of removing completely as it might be useful to test locally